FreeRTOS/Test/CMock/event_groups/event_groups_utest.c

```diff
@@ -0,0 +1,608 @@
+        pdTRUE,            /* BIT_0 & BIT_4 should be cleared before returning. */
+        pdTRUE,            /* Wait for both bits. */
+        xTicksToWait );    /* Wait a maximum of 100ms for either bit to be set. */
+
+    /* Validate: Expect to time-out and BitValue unchanged */
+    TEST_ASSERT_EQUAL( uxBitsSetVal, uxBitsGetVal );
+
+    /* Test-2: No bit was set */
+    uxBitsGetVal = xEventGroupWaitBits(
+        xEventGroupHandle, /* The event group being tested. */
+        BIT_0 | BIT_4,     /* The bits within the event group to wait for. */
+        pdFALSE,           /* BIT_0 & BIT_4 should not be cleared before returning. */
+        pdFALSE,           /* Don't wait for both bits, either bit will do. */
+        0 );               /* Don't block and wait */
+    /* Validate: Expect return and BitValue unchanged */
+    TEST_ASSERT_EQUAL( uxBitsSetVal, uxBitsGetVal );
+
+    /* Test-3: BIT_0 was set */
+    uxBitsSetVal = xEventGroupSetBits( xEventGroupHandle, BIT_0 );
+    uxBitsSetVal = xEventGroupWaitBits(
+        xEventGroupHandle, /* The event group being tested. */
+        BIT_0 | BIT_4,     /* The bits within the event group to wait for. */
+        pdFALSE,           /* BIT_0 & BIT_4 should not be cleared before returning. */
+        pdFALSE,           /* Don't wait for both bits, either bit will do. */
+        xTicksToWait );    /* Wait a maximum of 100ms for either bit to be set. */
+    uxBitsGetVal = xEventGroupGetBits( xEventGroupHandle );
+
+    /* Validate: Expect to return because BIT_0 was set */
+    TEST_ASSERT_EQUAL( BIT_0, uxBitsSetVal );
+    TEST_ASSERT_EQUAL( BIT_0, uxBitsGetVal & BIT_0 );
+
+    /* Test-4: BIT_0 and BIT_4 were set */
+    uxBitsSetVal = xEventGroupSetBits( xEventGroupHandle, BIT_4 );
+    uxBitsSetVal = xEventGroupWaitBits(
+        xEventGroupHandle, /* The event group being tested. */
+        BIT_0 | BIT_4,     /* The bits within the event group to wait for. */
+        pdTRUE,            /* BIT_0 & BIT_4 should be cleared before returning. */
+        pdTRUE,            /* Wait for both bits. */
+        xTicksToWait );    /* Wait a maximum of 100ms for either bit to be set. */
+    uxBitsGetVal = xEventGroupGetBits( xEventGroupHandle );
+
+    /* Validate: Expect to return because both BIT_0 and BIT_4 was set and then cleared */
+    TEST_ASSERT_EQUAL( BIT_0 | BIT_4, uxBitsSetVal & ( BIT_0 | BIT_4 ) );
+    TEST_ASSERT_EQUAL( 0, uxBitsGetVal & ( BIT_0 | BIT_4 ) );
+}
+
+
+/*!
+ * @brief validate tasks sync on event bits
+ * @coverage xEventGroupSync
+ */
+void test_xEventGroupSync_Success( void )
+{
+    /* Expectation of Function: xEventGroupCreate */
+    vListInitialise_Expect( 0 );
+    vListInitialise_IgnoreArg_pxList();
+    vListInitialise_ReturnThruPtr_pxList( pxListTemp );
+
+    /* Expectation of Function: xEventGroupSetBits x5 */
+    listGET_END_MARKER_ExpectAnyArgsAndReturn( ( ListItem_t * ) NULL );
```

Comment:
Please consider including list.c as a PROJECT_DEPS_SRC file rather than ignoring all of the arguments to the mocked list functions.

Reply:
so you means do not mock the list.c but include as dependency and then excludes it with the coverage filter?

Reply:
Files in PROJECT_DEPS_SRC do not get instrumented and don't product coverage data but do get built.

My main reason for this suggestion is that using mocks like this tightly couples the test to the implementation. Others might have a different idea on this.